src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -128,14 +135,46 @@ def init_from_file(self) -> None:
             profile_jsonc = commentjson.load(fileobj)
 
         self.profiles = profile_jsonc.get("profiles", {})
+        self.schema_property = profile_jsonc.get("$schema", None)
         self.defaults = profile_jsonc.get("defaults", {})
         self.schema_property = profile_jsonc.get("$schema", None)
 
+        if self.schema_property:
+            self.validate_schema(config_type, validate_schema, verify=verify)
         # loading secure props is done in load_profile_properties
         # since we want to try loading secure properties only when
         # we know that the profile has saved properties
         # self.load_secure_props()
 
+    def validate_schema(
+        self,
+        config_type: str,
+        validate_schema: Optional[bool] = False,
+        verify: Optional[bool] = False,
+    ) -> None:
+        """
+        Get the $schema_property from the config and load the schema
+
+        Returns
+        -------
+        file_path to the $schema property
+        """
+
+        path_schema_json, path_config_json = None, ""
+
+        if self._location:
+            path_config_json = f"./{self.filename}"
+
+        path_schema_json = self.schema_path
+        if path_schema_json is None:    # check if the $schema property is not defined
+            warnings.warn(
+                f"$schema property could not found"
+            )
+                
+        # validate the $schema property 
+        if path_schema_json and validate_schema:
+            validate_config_json(path_config_json, path_schema_json, cwd = self.location, verify=verify)
```

Comment:
Passing `path_config_json` as a string requires the file to be loaded from disk a 2nd time even though it has already been loaded into memory. Also sometimes the file may be loaded from the wrong directory.

Could we pass the contents of `zowe.config.json` as a JSON object instead? The JSON object could be stored on the class as `self.profile_jsonc` (see comment above)